test/new-e2e/tests/installer/unix/package_agent_test.go

```diff
@@ -461,3 +463,69 @@ func (s *packageAgentSuite) installDebRPMAgent() {
 	}
 
 }
+
+func (s *packageAgentSuite) TestInstallWithDDOT() {
+	// Install datadog-agent (base infrastructure)
+	s.RunInstallScript("DD_REMOTE_UPDATES=true", envForceInstall("datadog-agent"))
+	defer s.Purge()
+	s.host.AssertPackageInstalledByInstaller("datadog-agent")
+	s.host.WaitForUnitActive(s.T(), agentUnit, traceUnit)
+
+	// Install ddot
+	s.host.Run(fmt.Sprintf("sudo datadog-installer install oci://installtesting.datad0g.com/ddot-package:pipeline-%s", os.Getenv("E2E_PIPELINE_ID")))
+	s.host.AssertPackageInstalledByInstaller("datadog-agent-ddot")
+
+	// Check if datadog.yaml exists, if not return an error
+	s.host.Run("sudo test -f /etc/datadog-agent/datadog.yaml || { echo 'Error: datadog.yaml does not exist'; exit 1; }")
+
+	// Substitute API & site into otel-config.yaml
+	apiKey := os.Getenv("DD_API_KEY")
+	if apiKey == "" {
+		apiKey = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+	}
+	site := os.Getenv("DD_SITE")
+	if site == "" {
+		site = "datadoghq.com"
+	}
+	s.host.Run(fmt.Sprintf("sudo sh -c \"sed -i -e 's/\\${env:DD_API_KEY}/%s/' -e 's/\\${env:DD_SITE}/%s/' /etc/datadog-agent/otel-config.yaml\"", apiKey, site))
```

Comment:
Does `os.Getenv("DD_API_KEY/DD_SITE")` ever need to be used? If not, I think we can replace this block with the command used in `ddot_install_test`:
```
ExecuteWithoutError(t, VMclient, "sudo sh -c \"sed -i -e 's/\\${env:DD_API_KEY}/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/' -e 's/\\${env:DD_SITE}/datadoghq.com/' /etc/datadog-agent/otel-config.yaml\"")
```

I think we were trying to use env variables at first but after reading Baptiste's [comment](https://github.com/DataDog/datadog-agent/pull/38805#discussion_r2230857894) it seems like we don't need to.